lib/buttons-and-links/__tests__/KButton.spec.js

```diff
@@ -0,0 +1,67 @@
+import { shallowMount } from '@vue/test-utils';
+import KButton from '../KButton.vue';
+
+describe('KButton', () => {
+  describe('icon related props', () => {
+    it('should render an icon before the text with this.icon', () => {
```

Comment:
It took me a while to realize that you mean `icon` property with `this.icon`.

Might be a bit better readable if we used the same style like in one of the tests below that doesn't use `this` (`should render a dropdown icon when hasDropdown is true`), changing this description into `should render an icon before the text when icon is true`? Similarly in other tests that use `this`.

Reply:
I pushed updated strings